sax.h

```diff
@@ -0,0 +1,203 @@
+#ifndef SAX_H
+#define SAX_H
+
+#include "rapidjson/reader.h"
+#include "rapidjson/writer.h"
+#include "rapidjson/filereadstream.h"
+#include "rapidjson/filewritestream.h"
+#include "rapidjson/error/en.h"
+#include <cassert>
+#include <sstream>
+#include <fstream>
+#include <vector>
+#include <unordered_map>
+
+using namespace rapidjson;
+
+class Valuesax {
+public:
+    enum Type {
+        Decimal,
+        Double,
+        String,
+        Bool,
+        Null
+    };
+
+    static Valuesax Parse(std::string const& s) {
```

Comment:
please remove the static for Valuesax::parse 
no need for that

Reply:
But, I am calling parse functions in MyHandler class without instantiating an object of ValueSax

Reply:
using static is problematic.
The static function can modify only static data members.
it breaks the object-oriented paradigm.
the static data member is problematic, it's not thread-safe.
thus, we should avoid using static where possible.

there is no need for static in this case.

ValueSax::Parse is actually instantiating ValueSax in the function body, and returning it by value (it means to copy, not efficient)